sktime/datasets/tests/test_single_problem_loaders.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+"""Test single problem loaders with varying return types."""
+import numpy as np
+import pandas as pd
+import pytest
+
+from sktime.datasets import (  # Univariate; Unequal length; Multivariate
+    load_acsf1,
+    load_arrow_head,
+    load_basic_motions,
+    load_italy_power_demand,
+    load_japanese_vowels,
+    load_osuleaf,
+    load_plaid,
+    load_unit_test,
+)
+
+UNIVARIATE_PROBLEMS = [
+    load_acsf1,
+    load_arrow_head,
+    load_italy_power_demand,
+    load_osuleaf,
+    load_unit_test,
+]
+MULTIVARIATE_PROBLEMS = [
+    load_basic_motions,
+]
+UNEQUAL_LENGTH_PROBLEMS = [
+    load_plaid,
+    load_japanese_vowels,
+]
+
+
+def test_load_dataframe():
+    """Test that we can load all baked in TSC problems into nested pd.DataFrames."""
+    # should work for all
+    for loader in UNIVARIATE_PROBLEMS + MULTIVARIATE_PROBLEMS + UNEQUAL_LENGTH_PROBLEMS:
+        X, y = loader()
+        assert isinstance(X, pd.DataFrame)
+        assert isinstance(y, np.ndarray)
+        assert y.ndim == 1
+    for loader in UNIVARIATE_PROBLEMS + MULTIVARIATE_PROBLEMS + UNEQUAL_LENGTH_PROBLEMS:
+        X = loader(return_X_y=False)
+        assert isinstance(X, pd.DataFrame)
+
+
+def test_load_numpy3d():
```

Comment:
use `check_is_mtype`

Reply:
why?

Reply:
in order to have a single source of truth for data format specification, and to avoid duplication of code that carries out the same concern (checking compliance with said format specification)
https://www.oreilly.com/library/view/building-maintainable-software/9781491955987/ch04.html